src/plugins/layout.ts

```diff
@@ -166,12 +168,13 @@ export default class Layout implements IPlugin {
 
   private createEventState<T extends ILayoutEventInfo>(eventInfo: T): ILayoutState {
     let node = eventInfo.node;
-    let layoutState: ILayoutState = createLayoutState(node, this.shadowDom);
+    let styles = node.nodeType === Node.ELEMENT_NODE ? window.getComputedStyle(node as Element) : null;
```

Comment:
>let styles = node.nodeType === Node.ELEMENT_NODE ? window.getComputedStyle(node as Element) : null; [](start = 4, length = 99)

This is going to be expensive. Can we only do it for inserts and styles updates? Also, when stylesheet is inserted, would this mean you'd need to recompute styles for all nodes ?

Reply:
For correctness of box model, re-valuating dimensions is necessary and can't be behind a configuration. In other words box model as part of clarity should never be optional. 

Your point is valid, elements don't necessarily get a valid mutation event for CSS updates and therefore will need to be re-evaluated every time we get a mutation in the parent tree or addition of style sheet. What I can probably do to reduce the work here is to not capture box model for every element and instead limit collection to elements that we will need for the analysis.

Reply:
Do you need to evaluate all computed styles? Can you get only those you need by some alternate, faster means? Can we also profile the impact of this change before profile it? If we don't have it, let's add perf instrumentation to Clarity to make sure we can measure this kind of impact (for this change or any other).

Reply:
Unfortunately there's no faster way other than accessing all computed styles. E.g. element.style.visibility is much faster however it will only return a valid value if there's an inline style. For styles via CSS it doesn't work. 

Yes, added code for profiling. That was a good idea. On my machine we are averaging 50 operations per millisecond and I've made changes to only grab layout data for leaf nodes.